### ISSUE TITLE
ArangoDB causes logrotate to fail on Fedora 34 beta

### ISSUE THREAD

**Issue by hegjon** [arangodb/arangodb#13935]
## My Environment

* __ArangoDB Version__:        arangodb3-3.7.10-1.0.x86_64
* __Logrotate version__:        logrotate-3.18.0-2.fc34.x86_64
* __systemd version__:      systemd-248-1.fc34.x86_64
* __Infrastructure__:               Digital Ocean
* __Operating System__:        Fedora 34 beta

## Steps to reproduce

1. Install ArangoDB on Fedora 32
2. Upgrade to Fedora 34 beta
3. 

I know that Fedora 34 is still in beta, but we would like to be proactive and ready to upgrade after it is released in May.

__Problem__:

Logrotate fails with this error:
```
Apr 04 00:01:01 jonny-test1.zegeba.com systemd[1]: Starting Rotate log files...
Apr 04 00:01:01 jonny-test1.zegeba.com logrotate[14101]: error: arangodb3:6 unknown group '@LOGROTATE_GROUP@'
Apr 04 00:01:01 jonny-test1.zegeba.com logrotate[14101]: error: found error in /var/log/arangodb3/arangod.log , skipping
Apr 04 00:01:01 jonny-test1.zegeba.com logrotate[14101]: error: found error in file arangodb3, skipping
Apr 04 00:01:01 jonny-test1.zegeba.com systemd[1]: logrotate.service: Main process exited, code=exited, status=1/FAILURE
Apr 04 00:01:01 jonny-test1.zegeba.com systemd[1]: logrotate.service: Failed with result 'exit-code'.
Apr 04 00:01:01 jonny-test1.zegeba.com systemd[1]: Failed to start Rotate log files.
```

__Included logrotate config__:
```
cat /usr/share/arangodb3/arangodb3.logrotate
/var/log/arangodb3/arangod.log {
     rotate 12
     weekly
     compress
     delaycompress
     create 640  arangodb @LOGROTATE_GROUP@
     postrotate
     if rpm -q --quiet systemd ; then
       systemctl -q is-active arangodb3 && systemctl kill --signal=SIGHUP arangodb3 
     else
       /etc/init.d/arangodb3 reload-log > /dev/null
     fi
     endscript
     }

```

**Comment by Simran-B** [comment]
This doesn't look like an issue with ArangoDB to me, but rather a systemd/logrotate configuration problem.

Is `@LOGROTATE_GROUP@` logrotate's syntax for using the content of an environment variable?

Is this variable not set perhaps, in whatever context logrotate runs in?
If it doesn't get substituted it might be used verbatim, leading to `unknown group '@LOGROTATE_GROUP@'`.

**Comment by hegjon** [comment]
I think there is an issue during build where the variable is not set, since the others are set

https://github.com/arangodb/arangodb/blob/502eaf5bf9654e3447d6c59cf4d2c76bc81491f6/Installation/logrotate.d/arangod.systemd#L6

**Comment by Simran-B** [comment]
ArangoDB supports an `@key@` syntax in configuration files to use environment variable values:

https://www.arangodb.com/docs/stable/administration-configuration.html#environment-variables-as-parameters

arangod also has a startup option `--define` : Define key=value for a @key@ entry in config file

The file you linked wasn't changed in over 3 years, so I'm not sure what could have changed. @dothebart Do you have an idea? Could this be similar to the Debian locale issue? (the packaging processes spawned won't have a proper locale environment set)

**Comment by hegjon** [comment]
Looks like some other build scripts are setting the variable:

https://github.com/arangodb/arangodb/blob/e8027ee5d84dea3a621421f31aa5910d16f06d29/cmake/packages/packages.cmake#L31

https://github.com/arangodb/arangodb/blob/e8027ee5d84dea3a621421f31aa5910d16f06d29/cmake/packages/deb.cmake#L32

**Comment by Simran-B** [comment]
What happens if you modify `arangodb/Installation/logrotate.d/arangod.systemd` to use the user group name directly?

```diff
- create 640  arangodb @LOGROTATE_GROUP@ 
+ create 640  arangodb arangodb

**Comment by Simran-B** [comment]
@hegjon Can you try the above to see if that is a viable workaround? I'm not sure how one would set (environment) variables in the context of a logrotate config file.

**Comment by hegjon** [comment]
I confirm that applying your change suggestion on /etc/logrotate.d/arangodb3 on the server makes logrotate run without an error

**Comment by hegjon** [comment]
How can i replicate the build of the RPM?

**Comment by Simran-B** [comment]
@hegjon Have a look here at our build automation:

https://github.com/arangodb/oskar/blob/2ca20f9064bcdfc57323eef6b465b6324bfbf38e/helper.linux.fish#L676-L704

**Comment by KVS85** [comment]
@hegjon the fix will appear in RPM packages for 3.7.17 and 3.8.5.
